File: spark-sample/src/main/java/com/mh13/spark/SparkSample.java
/**
 * 
 */
package com.mh13.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

/**
 * @author amola
 *
 */
public class SparkSample {

	/**
	 * @param args
	 */
	
	public static void main(String[] args) {
		SparkConf conf = new SparkConf().setAppName("first_spark").setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);
		JavaRDD<String> lines =  sc.textFile("/Tools/spark.txt");
		lines.map(new Function<String, Integer>() {
			public Integer call(String line) throws Exception {
				return Integer.parseInt(line)*100;
			}
		}).saveAsTextFile("file:///Tools/spark_out.txt");

	}
	
}

File: spark-sample-scala/src/main/scala/SparkSample.scala
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object SparkSample extends App{
	val conf = new SparkConf().setAppName("spark-sample-app");
	val sc = new SparkContext(conf)
	println("SparkContext created")

	val lines = sc.textFile("file:///Users/amol/tmp/data/sample.txt")

	val words = lines.flatMap(_.split(" ")) 

	val counts = words.map(word => (word,1)).reduceByKey{case (x,y) => x+y}

	counts.saveAsTextFile("file:///Users/amol/tmp/data/sample_output.txt")
}